app/scripts/modules/core/src/pipeline/config/validation/baseRequiredField.validator.ts

```diff
@@ -0,0 +1,56 @@
+import { get, has } from 'lodash';
+
+import { IPipeline, IStage, IStageOrTriggerTypeConfig, ITrigger } from 'core/domain';
+import { IStageOrTriggerValidator, IValidatorConfig } from './pipelineConfig.validator';
+
+export interface IRequiredField {
+  fieldName: string;
+  fieldLabel?: string;
+}
+
+export interface IBaseRequiredFieldValidationConfig extends IValidatorConfig {
+  message?: string;
+}
+
+export abstract class BaseRequiredFieldValidator implements IStageOrTriggerValidator {
+  public validate(
+    pipeline: IPipeline,
+    stage: IStage | ITrigger,
+    validationConfig: IBaseRequiredFieldValidationConfig,
+    config: IStageOrTriggerTypeConfig,
+  ): string {
+    if (!this.passesValidation(pipeline, stage, validationConfig)) {
+      return this.validationMessage(validationConfig, config);
+    }
+    return null;
+  }
+
+  protected abstract passesValidation(
+    pipeline: IPipeline,
+    stage: IStage | ITrigger,
+    validationConfig: IBaseRequiredFieldValidationConfig,
+  ): boolean;
+
+  protected abstract validationMessage(
+    validationConfig: IBaseRequiredFieldValidationConfig,
+    config: IStageOrTriggerTypeConfig,
+  ): string;
+
+  protected printableFieldLabel(field: IRequiredField): string {
+    let fieldLabel: string = field.fieldLabel || field.fieldName;
+    return fieldLabel.charAt(0).toUpperCase() + fieldLabel.substr(1);
```

Comment:
this is also available as a lodash built-in: https://lodash.com/docs/4.17.10#upperFirst